internal/api/router/manager.go

```diff
@@ -34,24 +34,35 @@ func NewManager(deviceRepo DevicesRepository, reports *report.Repository) *Manag
 }
 
 // ListenAndServe starts to serve Web API requests.
-func (m *Manager) ListenAndServe(ctx context.Context, address string, port int) {
+func (m *Manager) ListenAndServe(ctx context.Context, address string, port int) error {
 	defer func() {
 		log.Warn().Msg("Shutdown.")
 	}()
 
 	tlsConfig := &tls.Config{
 		InsecureSkipVerify: config.Cfg.LDAP.InsecureSkipVerify, //nolint:gosec // configurable on purpose
 	}
-	ldap := auth.NewLDAPAuth(config.Cfg.LDAP.URL, config.Cfg.LDAP.BindDN, config.Cfg.LDAP.Password, config.Cfg.LDAP.BaseDN, tlsConfig)
+
+	var withAuth auth.BasicAuth
+	if config.Cfg.LDAP.Enabled {
+		withAuth = auth.NewBasicAuth(auth.LDAPMode)
+
+		ldap := auth.NewLDAPAuth(config.Cfg.LDAP.URL, config.Cfg.LDAP.BindDN, config.Cfg.LDAP.Password, config.Cfg.LDAP.BaseDN, tlsConfig)
+		if err := withAuth.ConfigureLdap(ldap); err != nil {
```

Comment:
Could we pass the configuration directly into `NewBasicAuth` method, so we would not use two functions to initialize the authenticator ?

Reply:
done